pool_func.go

```diff
@@ -165,161 +64,42 @@ func NewPoolWithFunc(size int, pf func(interface{}), options ...Option) (*PoolWi
 // Pool.Invoke() call once the current Pool runs out of its capacity, and to avoid this,
 // you should instantiate a PoolWithFunc with ants.WithNonblocking(true).
 func (p *PoolWithFunc) Invoke(args interface{}) error {
-	if p.IsClosed() {
-		return ErrPoolClosed
-	}
-	var w *goWorkerWithFunc
-	if w = p.retrieveWorker(); w == nil {
-		return ErrPoolOverload
-	}
-	w.args <- args
-	return nil
+	return p.pool.Submit(func() {
```

Comment:
This approach creates a new anonymous function for every call to `Invoke()`, it could result in a noticeable GC pressure when there are a massive amount of calls to `Invoke()`.

Reply:
After writing similar benchmarks to those already in place for PoolFunc, and profiling those benches, there doesn't seem to be much contention going on for 1M tasks being sent to 200K workers. Although I do agree that this probably isn't the best way of profiling such a change, it seems promising. Here's the bench results:

```
go test -bench=. -run=Bench -benchtime=10s -benchmem=true -cpuprofile cpu.out -memprofile mem.out                                  

goos: linux
goarch: amd64
pkg: github.com/panjf2000/ants/v2
cpu: 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
BenchmarkGoroutines-8               	      19	 563469615 ns/op	97110809 B/op	 2009954 allocs/op
BenchmarkSemaphore-8                	      14	 790446160 ns/op	104708941 B/op	 2007159 allocs/op
BenchmarkAntsPool-8                 	      13	 914571429 ns/op	17268787 B/op	 1016886 allocs/op
BenchmarkAntsPoolFunc-8             	      12	 986515902 ns/op	33669592 B/op	 1021049 allocs/op
BenchmarkGoroutinesThroughput-8     	      14	 874425709 ns/op	98912156 B/op	 1074689 allocs/op
BenchmarkSemaphoreThroughput-8      	      15	 820485970 ns/op	96008005 B/op	 2000036 allocs/op
BenchmarkAntsPoolThroughput-8       	      14	 946832862 ns/op	  807021 B/op	   12685 allocs/op
BenchmarkAntsPoolFuncThroughput-8   	      13	 931821087 ns/op	32611812 B/op	 1009524 allocs/op
```

Would you mind giving your input on the profiling data?